File: 11. Container With Most Water/containerWithMostWater.ts
function maxArea(height: number[]): number {
    let maxArea = 0;
    let left = 0;
    let right = height.length - 1;
    while (left < right) {
        let width  = right - left
        let minHeight = Math.min(height[left], height[right])
        if (maxArea < minHeight * width) {
            maxArea = minHeight * width
        }
        if (height[left] > height[right]) {
            right--
        }
        else if (height[left] < height[right]) {
            left++
        }
        else {
            right--
            left++
        }
    } 
    return maxArea
};

File: 11. Container With Most Water/containerWithMostWater.py
def container_with_most_water(height: list[int]) -> int:
    max_area: int = 0
    left: int = 0
    right: int = len(height) - 1
    while left < right:
        width = right - left
        min_height = min(height[right], height[left])

        if max_area < width * min_height:
            max_area = width * min_height

        if height[right] > height[left]:
            left += 1
        elif height[right] < height[left]:
            right -= 1
        else:
            left += 1
            right -= 1 
    
    return max_area
